RFC/src/RFC-0241_AtomicSwapXMR.md

```diff
@@ -0,0 +1,830 @@
+(xm_a + xm_b) \cdot G_m &= Xm_a + Xm_b \\\\
+(xm_a + xm_b) \cdot G_m &= Xm \\\\
+\end{aligned}
+\tag{2}
+$$
+
+We know that \\(Xm\\), \\(Xm_a\\), \\(Xm_b\\) are public. While \\(xm\\), \\(xm_a\\), \\(xm_b\\) are all private.
+
+But if we expose \\(xm_b\\), we can try to do the following:
+$$
+\begin{aligned}
+(xm_a + xm_b) \cdot G_m &= Xm_a + Xm_b\\\\
+xm_a \cdot G_m &= (Xm_a + Xm_b - xm_b \cdot G_m) \\\\
+xm_a \cdot G_m &= Xm_a \\\\
+\end{aligned}
+\tag{3}
+$$
+
+However, this is the Elliptic-Curve Discrete Logarithm Problem, and there is no easy solution to solve this on current computer
+hardware. Thus this is still secure even though we leaked part of the secret key \\(xm\\).
+
+### TariScript
+
+The Script used for the Tari UTXO is as follows:
+``` TariScript,ignore
+   ConvertToEd25519Point
+   CheckHeight(height_1)
+   LtZero
+   IFTHEN
+      PushHash(Xm_b)
+      EqualVerify
+      HashSha256 
+      PushHash(HASH256{pre_image})
+      EqualVerify
+      PushPubkey(K_{Sb})
+   Else
+      CheckHeight(height_2)
+      LtZero
+      IFTHEN
+         PushHash(Xm_a)
+         EqualVerify
+         PushPubkey(K_{Sa})
+      Else
+         PushHash(Xm_b)
+         EqualVerify
+         PushPubkey(K_{Sb})
+      ENDIF
+   ENDIF
+```
+
+Before `height_1`, Bob can claim the Tari UTXO by supplying `pre_image` and his private Monero key part `x_b`. After 
+`height_1` but before `height_2`, Alice can claim the Tari UTXO by supplying her private Monero key part `x_a`. After 
+`height_2`, Bob can claim the Tari UTXO by providing his private Monero key part `x_b`.
+The operation `PushHash` is used to push the Ed25519 public key onto the stack, we don't want to use the public key here except for
+comparison purposes, so we keep it as a byte array. 
+
+### Negotiation
+
+Alice and Bob have to negotiate the exchange rate and the amount exchanged in the atomic swap. They also need to decide 
+how the two UTXO's will look on the blockchain. To accomplish this, the following needs to be finalized:
+
+* Amount of Tari to swap for the amount of Monero
+* Monero public key parts \\(Xm_a\\), \\(Xm_b\\) and its aggregate form \\(Xm\\)
+* Tari [script key] parts \\(K_{Sa}\\), \\(K_{Sb}\\) 
+* The [TariScript] to be used in the Tari UTXO
+* The blinding factor \\(k_i\\) for the Tari UTXO, which can be a Diffie-Hellman between their addresses.
+
+
+### Key selection
+
+Using (1), we create the Monero keys as they are multi-party aggregate keys.
+The Monero key parts for Alice and Bob is constructed as follows:
+
+
+$$
+\begin{aligned}
+Xm_a' &= xm_a' \cdot G_m \\\\
+Xm_b' &= xm_b' \cdot G_m \\\\
+xm_a &=  \hash{\hash{Xm_a' \cat Xm_b'} \cat Xm_a' } * xm_a' \\\\
+xm_b &=  \hash{\hash{Xm_a' \cat Xm_b'} \cat Xm_b' } * xm_b' \\\\
+Xm_a &=  \hash{\hash{Xm_a' \cat Xm_b'} \cat Xm_a' } * Xm_a' \\\\
+Xm_b &=  \hash{\hash{Xm_a' \cat Xm_b'} \cat Xm_b' } * Xm_b' \\\\
+xm &= xm_a + xm_b \\\\
+\end{aligned}
+\tag{4}
+$$
+
+
+### Commitment phase
```

Comment:
I think it is important to mention here why a DLEQ proof is not needed and how the script (committed to in the input script signature) fulfils that role, especially to prime the audience that expects a DLEQ proof should be present.